authors/apps/articles/tests/article_tests_base_class.py

```diff
@@ -46,3 +46,4 @@ def setUp(self):
         self.non_existent_article = '/api/articles/hi/rate/'
         self.get_non_existent_article = '/api/articles/hi/ratings/'
         self.article_rating = '/api/articles/'
+        self.bookmark_article_url = '/api/articles/{}/bookmark/'.format(self.slug)
```

Comment:
Awesome job, I like how you have made this test URL dynamic. May I suggest using the  `reverse` function from `django.urls` here. It makes the entire URL dynamic so in case the URLs change the we will avoid the pain of updating tests.
```suggestion
        self.bookmark_article_url = reverse("articles:create_bookmark", args=[self.slug])
```

Reply:
I will look into the reverse function and make the proposed changes.